boe_min.sql

```diff
@@ -6,81 +6,80 @@
+    con_mwbe INTEGER NOT NULL,  -- 0 for no, 1 for yes; allows further specification into the future (e.g., 0 for no, 1+ for different kinds)
     CONSTRAINT con_type_id_fk FOREIGN KEY (con_type_id) REFERENCES contractor_types(con_type_id),
     CONSTRAINT con_acc_id_fk FOREIGN KEY (con_acc_id) REFERENCES accounts (acc_id)
 );
 
 CREATE TABLE IF NOT EXISTS contractor_types(
-    con_type_id INTEGER CONSTRAINT con_type_id_pk PRIMARY KEY AUTOINCREMENT,
-    con_type_name TEXT NOT NULL
+    con_type_id INTEGER CONSTRAINT con_type_id_pk PRIMARY KEY AUTOINCREMENT,  -- id number for contractor type
+    con_type_name TEXT NOT NULL  -- what the contractor type is (for human-readable outputs)
 );
 
 CREATE TABLE IF NOT EXISTS departments(
-    dept_id INTEGER CONSTRAINT dept_id_pk PRIMARY KEY AUTOINCREMENT,
-    dept_name TEXT NOT NULL
+    dept_id INTEGER CONSTRAINT dept_id_pk PRIMARY KEY AUTOINCREMENT,  -- dept id number
+    dept_name TEXT NOT NULL  -- name of the department (for human-readable outputs)
 );
 
 CREATE TABLE IF NOT EXISTS applicants(
-    appl_id INTEGER CONSTRAINT app_id_pk PRIMARY KEY AUTOINCREMENT,
-    appl_name TEXT NOT NULL,
-    appl_address TEXT NOT NULL,
-    appl_con_id INTEGER,
+    appl_id INTEGER CONSTRAINT app_id_pk PRIMARY KEY AUTOINCREMENT,  -- id number for applicants
+    appl_name TEXT NOT NULL,  -- name of the person/org filing the application (they may not be a contractor)
+    appl_address TEXT NOT NULL,  -- address of the person/org filing the application
+    appl_con_id INTEGER,  -- the contractor id (if applicable, contractors can file applications too)
     CONSTRAINT appl_con_id_fk FOREIGN KEY (appl_con_id) REFERENCES contractors (con_id)
 );
 
 CREATE TABLE IF NOT EXISTS permits(
-    perm_id INTEGER CONSTRAINT perm_id_pk PRIMARY KEY AUTOINCREMENT,
-    perm_address TEXT NOT NULL,
-    perm_payment INTEGER NOT NULL,
-    perm_date TEXT,
-    perm_grant_date TEXT NOT NULL,
-    perm_appl_id INTEGER NOT NULL,
-    perm_desc TEXT NOT NULL,
-    perm_notes TEXT,
-    perm_objections TEXT,
+    perm_id INTEGER CONSTRAINT perm_id_pk PRIMARY KEY AUTOINCREMENT,  -- id number for the permit
+    perm_address TEXT NOT NULL,  -- the address/location the permit is for
+    perm_payment INTEGER NOT NULL,  -- the total cost of the permit
+    perm_date TEXT,  -- the date the permit is for
+    perm_grant_date TEXT NOT NULL,  -- the date the permit was granted
+    perm_appl_date TEXT NOT NULL,  -- the date the permit was applied for
+    perm_appl_id INTEGER NOT NULL,  -- the id number of the applicant (see TABLE applicants)
+    perm_desc TEXT NOT NULL,  -- the text of the permit
+    perm_notes TEXT,  -- any additional notes for this permit
+    perm_objections TEXT,  -- any objections to the denial/issuance of the permit
     CONSTRAINT perm_appl_id_fk FOREIGN KEY (appl_id) REFERENCES applicants (appl_id)
 );
 
 CREATE TABLE IF NOT EXISTS prequal(
```

Comment:
So each row here represents an act of prequalilification, allowing us to represent cases where the same contractor gets prequalified multiple times.